presto-kafka/src/main/java/com/facebook/presto/kafka/KafkaSplitManager.java

```diff
@@ -129,6 +138,22 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
         }
     }
 
+    private static String readSchema(String dataSchemaLocation)
+    {
+        try {
+            InputStream inputStream = isURL(dataSchemaLocation.trim().toLowerCase(Locale.ENGLISH)) ? new URL(dataSchemaLocation).openStream() : new FileInputStream(new File(dataSchemaLocation));
+            return CharStreams.toString(new InputStreamReader(inputStream, UTF_8));
+        }
+        catch (IOException e) {
+            throw new PrestoException(GENERIC_INTERNAL_ERROR, "Could not parse the AVRO schema at: " + dataSchemaLocation);
+        }
+    }
+
+    private static boolean isURL(String location)
```

Comment:
Is there a reason why you don't use the `URI` constructor and handle the exception if it's not a url, etc.?